### PR TITLE
Restructure tests directory hierarchy (pytest)

### PR DESCRIPTION
**Pull request by irenaby**
## Pull Request Description:
Restructure pytest directory into two levels: first level is framework/common, second level is unit/integration.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: tests_pytest/keras_tests/keras_test_util/keras_test_mixin.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import numpy as np
+
+from model_compression_toolkit.core.keras.default_framework_info import DEFAULT_KERAS_INFO
+from model_compression_toolkit.core.keras.keras_implementation import KerasImplementation
+from model_compression_toolkit.target_platform_capabilities.targetplatform2framework.attach2keras import \
+    AttachTpcToKeras
+
+
+class KerasFwMixin:
+    """ Mixin helper class containing keras-specific definitions. """
```
Comment (ofirgo) [inline tests_pytest/keras_tests/keras_test_util/keras_test_mixin.py:26]: What's Mixin?

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: tests_pytest/common_tests/unit_tests/core/common/__init__.py
[the remark this answers is not in the document]
Comment (irenaby) [reply tests_pytest/common_tests/unit_tests/core/common/__init__.py:1]: its core/common, same as in the code

File: tests_pytest/keras_tests/keras_test_util/keras_test_mixin.py (hunk 1 above)
Comment (irenaby) [reply tests_pytest/keras_tests/keras_test_util/keras_test_mixin.py:26]: A design patter that provides specific functionality via class. It's not used in this PR but you can do
class MyTest(MaybeSomeOtherClass, KerasFwMixin) and get the functionality. The purpose of inheritance here is just to inject functionality.

**Review by ofirgo — APPROVED** [review]

File: tests_pytest/common_tests/unit_tests/core/common/__init__.py
[the remark this answers is not in the document]
Comment (irenaby) [reply tests_pytest/common_tests/unit_tests/core/common/__init__.py:1]: removed
